design/feature-flags.md

```diff
@@ -0,0 +1,98 @@
+# Feature Flags
```

Comment:
This doc, on its own, does not capture the final user experience of how feature flags will be defined and then consumed.

I got this this from the changes to the `CONTRIBUTING.md` but read on its own (typically, when read in posterity and after this PR is merged), this doc is fairly vague.

Reply:
Yeah, I agree. Some sample code in this doc would help I think.

Reply:
I'll try to incorporate the section from CONTRIBUTING into the RFC.

Reply:
Done.